packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -0,0 +1,157 @@
+};
+
+const createDataType = (type: FieldType): ts.Node => {
+  if (type.kind === 'Scalar') {
+    return ts.factory.createCallExpression(
+      ts.factory.createIdentifier(`${REFERENCE_A}.${getTypescriptDataSchemaType(type.name)}`),
+      undefined,
+      undefined,
+    );
+  }
+
+  if (type.kind === 'Enum') {
+    return ts.factory.createCallExpression(ts.factory.createIdentifier(`${REFERENCE_A}.${ENUM_METHOD}`), undefined, [
+      ts.factory.createArrayLiteralExpression(
+        type.values.map((value) => ts.factory.createStringLiteral(value)),
+        true,
+      ),
+    ]);
+  }
+
+  // We do not import any Database type as 'Custom' type.
+  // In case if there is a custom type in the IR schema, we will import it as string.
+  if (type.kind === 'Custom') {
+    return ts.factory.createCallExpression(
+      ts.factory.createIdentifier(`${REFERENCE_A}.${STRING_METHOD}`),
+      undefined,
+      undefined,
+    );
+  }
+
+  // List or NonNull
+  const modifier = type.kind === 'List' ? ARRAY_METHOD : REQUIRED_METHOD;
+  return ts.factory.createCallExpression(
+    ts.factory.createPropertyAccessExpression(createDataType(type.type) as ts.Expression, ts.factory.createIdentifier(modifier)),
+    undefined,
+    undefined,
+  );
+};
+
+const getTypescriptDataSchemaType = (type: string): string => {
+  const DEFAULT_DATATYPE = STRING_METHOD;
+  const tsDataSchemaType = GQL_TYPESCRIPT_DATA_SCHEMA_TYPE_MAP[type.toLowerCase()];
+  return tsDataSchemaType ?? DEFAULT_DATATYPE;
+};
+
+const createModelDefinition = (model: Model): ts.Node => {
+  const properties = model.getFields().map((field) => {
+    return createProperty(field);
+  });
+  return ts.factory.createObjectLiteralExpression(properties as ts.ObjectLiteralElementLike[], true);
+};
+
+const createModel = (model: Model): ts.Node => {
+  const modelExpr = ts.factory.createCallExpression(
+    ts.factory.createIdentifier(`${REFERENCE_A}.${MODEL_METHOD}`),
+    undefined,
+    [createModelDefinition(model) as ts.Expression],
+  );
+
+  const modelExprWithPK = ts.factory.createCallExpression(
+    ts.factory.createPropertyAccessExpression(modelExpr, ts.factory.createIdentifier(IDENTIFIER_METHOD)),
+    undefined,
+    [
+      ts.factory.createArrayLiteralExpression(
+        model
+          .getPrimaryKey()
+          .getFields()
+          .map((field) => ts.factory.createStringLiteral(field)),
+        true,
+      ),
+    ],
+  );
+
+  return ts.factory.createPropertyAssignment(ts.factory.createStringLiteral(model.getName()), modelExprWithPK as ts.Expression);
+};
+
+const exportModifier = ts.factory.createModifier(ts.SyntaxKind.ExportKeyword);
+
+/**
+ * Creates a typescript data schema from internal SQL schema representation
+ * Example typescript data schema:
+ * ```
+ * export const schema = a.schema({              <--- createSchema()
+ *   User: a.model({                             <--- createModel()
+ *     id: a.string().required(),                <--- createProperty()
```

Comment:
This is really handy -- can you extend it to the lines below? I'm wondering where things like `createDataType` come in.